### PR TITLE
chore: Remove Embedded Social

### LINKED ISSUE #827: ES - Unnecessary repo dependency?
**Summary:** 

Looking at the main `build.gradle` in the `repositories` block, we have a Jcenter repo:

~~~
    maven {
        // for testlib
        url "http://dl.bintray.com/acrowntest/test"
    }
~~~

@acrown-msft Is this still needed for Embedded Social?  Should it be removed at some point in the future?

**Steps to reproduce:** 

Look at `build.gradle` in the `repositories` block.

### LINKED ISSUE #852: ES - Multiple build flavors use the same Google OAuth credentials
**Summary:** 

If you install more than one build flavor of OBA (e.g., `obaGoogle` and `obaAgencyX`), and then try to log in with embedded social, you get prompted to associate the login with more than one application.  I believe this is due to the hard-coded value `com.joulespersecond.seattlebusbot` for `package_name` in `onebusaway-android/google-services.json`.  This should be dynamically changed to reflect the `applicationId` defined in `build.gradle`.  For example, for Agency X it should be `org.agencyx.android`.

Screenshot:

image

Related to https://github.com/OneBusAway/onebusaway-android/issues/853, as to fully support independent ES profiles per build flavor, each build flavor also needs it's own Embedded Social API key.

**Steps to reproduce:** 

1. Install both the `obaGoogle` and `obaAgencyX` build flavors
1. Open `obaGoogle` (i.e., normal) OBA app
1. Go to navigation drawer and tap "Log in"
1. Tap "Google"

**Expected behavior:** 

Take me straight to the Google OAuth screen on behalf of the normal OBA app

**Observed behavior:** 

It gives me the option to handle the action using either the normal OBA app or the Agency X app

**Device and Android version:** 

Samsung Galaxy S8+ w/ Android 8.0 beta

**Screenshots:** 

See above

### LINKED ISSUE #870: IllegalStateException in ES SDK onTokenRequestCompleted()
**Summary:** 

Issue documented in Embedded Social SDK - https://github.com/Microsoft/EmbeddedSocial-Android-SDK/issues/88 - it needs to first be fixed there.

From the OneBusAway Android rollout via the Android Developer Console - affecting 3 users (3 crashes):

~~~
java.lang.IllegalStateException: 
  at android.support.v4.app.Fragment.getResources (Fragment.java:608)
  at android.support.v4.app.Fragment.getString (Fragment.java:632)
  at com.microsoft.embeddedsocial.c.d$1.onTokenRequestCompleted (Unknown Source:30)
  at net.openid.appauth.AuthorizationService$TokenRequestTask.onPostExecute (AuthorizationService.java:308)
  at net.openid.appauth.AuthorizationService$TokenRequestTask.onPostExecute (AuthorizationService.java:238)
  at android.os.AsyncTask.finish (AsyncTask.java:695)
  at android.os.AsyncTask.-wrap1 (Unknown Source)
  at android.os.AsyncTask$InternalHandler.handleMessage (AsyncTask.java:712)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6940)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)
~~~

I'm assuming this happens when someone tries to log into Embedded Social in the OBA app and then closes it, likely under poor wireless conditions that cause a significant delay in onTokenRequestCompleted() returning.

**Steps to reproduce:** 

Unknown

**Expected behavior:** 

Not crash

**Observed behavior:** 

Crash

**Device and Android version:** 

Devices affected:
* Samsung Galaxy Note8 (greatqlte), 6144MB RAM, Android 8.0
* Google Pixel 2 (walleye), 4096MB RAM, Android 8.1
* ZTE Blade X Max (stollen), 2048MB RAM, Android 7.1

Note that the Note 8 and Pixel 2 are affected, so this does affect flagship devices

### LINKED ISSUE #887: Add Embedded Social to Nearby (map) view
**Summary:** 

An interesting app called "Pigeon" surfaced recently, and it shows social posts in context of transit for NYC (still invite-only for iOS):
https://www.theverge.com/2018/5/31/17414330/google-area-120-pigeon-nyc-subway-app

They have an interesting map view that shows social posts as pigeon markers:

image

We could do something similar with Embedded Social on OBA, such as showing red dots on hot stops, or after we have user-generated topics the topics themselves could be geo-tagged with their own icon.

Right now this is blocked by a working "Popular" topics implementation for stop and route-specific topics.

**Steps to reproduce:** 

1. Go to Nearby view

**Expected behavior:** 

Show me something that indicates where hot stops/routes/topics are located

**Observed behavior:** 

We don't show any Embedded Social info in context of the map

### LINKED ISSUE #888: Add Embedded Social to trip status/details view
**Summary:** 

An interesting app called "Pigeon" surfaced recently, and it shows social posts in context of transit for NYC (still invite-only for iOS):
https://www.theverge.com/2018/5/31/17414330/google-area-120-pigeon-nyc-subway-app

They have an interesting trip status view that shows social comments on stops along a trip:

image

We could do something similar with Embedded Social on OBA, such as showing comments in-line the trip status/details for stops and/or trips/routes.

Right now this is potentially blocked by a working "Popular" topics implementation for route-specific topics, although potentially we could allow users to generate comments per-trip on-demand.

**Steps to reproduce:** 

1. Tap on arrival time
1. Tap on "Show trip status"

**Expected behavior:** 

Show me user-generated content (comments) for the trip/route or stops in the trip status view

**Observed behavior:** 

There is no Embedded Social information in the trip status view

### LINKED ISSUE #889: Social "Popular" feed is empty
**Summary:** 

The "Popular" topics section of the navigation drawer is always empty.

**Steps to reproduce:** 

Open the popular topics feed from the navigation drawer.

**Expected behavior:** 

Display popular stops and routes.

**Observed behavior:** 

UI displays an empty feed.

Current solution will be to remove this item from the Navigation Drawer until an Embedded Social server-side fix is ready.

### PR DESCRIPTION
**Pull request by acrown-msft**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `gradlew connectedObaGoogleDebugAndroidTest connectedObaAmazonDebugAndroidTest` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them for the initial submission of the pull request.  When addressing comments on a pull request, please push a new commit per comment when possible (reviewers will squash and merge using GitHub merge tool)

Closes #889, Closes #888, Closes #887, Closes #870, Closes #852, Closes #827

### CODE REVIEW

File: onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java [hunk 1]
```diff
@@ -82,28 +80,15 @@
...
 
     protected static final int NAVDRAWER_ITEM_PLAN_TRIP = 6;
-    
-    // 7 is reserved for re-adding the Embedded Social "Popular" item - See #889
-    //protected static final int NAVDRAWER_ITEM_POPULAR = 7;
 
-    protected static final int NAVDRAWER_ITEM_PINS = 8;
+    protected static final int NAVDRAWER_ITEM_OPEN_SOURCE = 7;
 
-    protected static final int NAVDRAWER_ITEM_ACTIVITY_FEED = 9;
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java:91]: So what happens if someone has one of the ES screens as their most recently used screen, installs the update, and then starts the app? Seems like currently some strange behavior would happen. 

I suggest we leave all these ES `int` values here, mark them as deprecated, and then handle them in this class and `HomeActivity` by loading up the `NAVDRAWER_ITEM_NEARBY` when ES screen was selected. That way someone who previously has an ES screen as most recently used would then end up in the "Nearby" view.

File: onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java [hunk 2]
```diff
@@ -310,7 +280,7 @@ public void run() {
      */
     public void selectItem(int position) {
         setSelectedNavDrawerItem(position);
-        if (mDrawerLayout != null && mFragmentContainerView != null) {
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java:313]: I would leave this null check in place.

File: onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanActivity.java [hunk 3]
```diff
@@ -259,16 +258,6 @@ public void onBackPressed() {
         }
     }
 
-    @Override
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanActivity.java:262]: I would leave this in place - seems like it should work like this regardless of ES.

File: onebusaway-android/src/main/java/org/onebusaway/android/util/RegionUtils.java [hunk 4]
```diff
@@ -464,7 +464,6 @@ public static String formatOtpBaseUrl(String baseUrl) {
                         c.getString(12),             // OTP Base URL
                         c.getString(13),              // OTP Contact Email
                         c.getInt(14) > 0,           // Supports Otp Bikeshare
-                        c.getInt(15) > 0,            // Supports Embedded Social
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/util/RegionUtils.java:467]: Per previous comment, let's leave the Regions API response intact in the app for now.

**Review by barbeau — CHANGES_REQUESTED** [review]
@acrown-msft Thanks for working on this! It mostly looks good from a look at the code in the PR, although a few requests in-line. I still need to do a test install and upgrade with these changes, but I figured I'd try to get you some feedback sooner rather than later.

File: onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java (hunk 1 above)
Comment (acrown-msft) [reply onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java:91]: Good catch; these values need to be reserved.

**Review by barbeau — APPROVED** [review]

**Comment by barbeau** [comment]
Thanks @acrown-msft!

**Pull request merged**